tests/datatypes/test_str.py

```diff
@@ -356,6 +356,19 @@ def test_ljust(self):
                 print(err)
                 """)
 
+    def test_partition(self):
+        self.assertCodeExecution("""
+            s = "foobar"
+            print(s.partition("ob"))
+            print(s.partition("o"))
+            print(s.partition("f"))
+            print(s.partition("r"))
```

Comment:
This looks really good - but it's missing one obvious test - partitioning on a substring that doesn't exist.